packages/preferences/src/browser/views/preference-searchbar-widget.tsx

```diff
@@ -139,7 +143,7 @@ export class PreferencesSearchbarWidget extends ReactWidget implements StatefulW
         const optionContainer = this.renderOptionContainer();
         return (
             <div className='settings-header'>
-                <div className="settings-search-container">
+                <div className="settings-search-container" ref={this.focus}>
```

Comment:
Question: Is there a meaningful difference between `ref={this.focus}` + `this.focus = this.focus.bind(this);` and `ref={() => this.focus}`? If not, we should always prefer to use the latter. The former is quite unintuitive and overcomplicates the code in my opinion.

Reply:
Yes, unfortunately there's a difference. When using an arrow function like this `ref={() => this.focus}`, it will have a new identity on each re-render. The `ref` will be recreated each time, triggering two calls to the callback: first when the `ref` will be set to null and again when the function will be assigned to the `ref`. So on every render (e.g. on each key press in the search bar), the `focus` function will be called twice.

This does not happen when using a stable callback, like it is the case currently in the code.